modules/service/src/main/resources/db/migration/V0843__program_reference.sql

```diff
@@ -0,0 +1,90 @@
+-- Add a semester column to program.  It is nullable but there will be a
+-- constraint that checks that it is set when the proposal status is not
+-- 'not_submitted'.
+
+CREATE DOMAIN d_semester AS varchar
+  CHECK (VALUE ~ '^[0-9][0-9][0-9][0-9]+[AB]$');
```

Comment:
Is there any practical reason this can't be `'^20[0-9][0-9][AB]$'`? I can't imagine any of this will still be running in 75 years.

Reply:
In fact I guess we could constrain the year to be 2024 through current year + 1, with a looser constraint on the Scala side.

Reply:
Thanks, that's a good idea.  I'll make the lower limit be 2000 though since I think eventually we may want to import old data in some form?